crates/router/src/services/api.rs

```diff
@@ -1328,6 +1328,11 @@ impl EmbedError for Report<api_models::errors::types::ApiErrorResponse> {
     }
 }
 
+impl EmbedError
```

Comment:
I think we can remove this entire trait, we don't use it anywhere at all, from what I know.
This can be taken up in a separate PR, please create an issue and mark it as a good first issue.